returnn/tf/layers/basic.py

```diff
@@ -2685,6 +2685,83 @@ def get_out_data_from_opts(cls, name, axis, dim=1, sources=(), **kwargs):
     return data
 
 
+class RepeatLayer(_ConcatInputLayer):
+  """
+  A wrapper around tf.repeat, but supports an additional batch axis for the durations
+  The sum of the repetitions has to be non-zero for each sequence in the batch.
+
+  This layer can only be used with Tensorflow 1.15.0 or newer.
+  """
+  layer_class = "repeat"
+
+  def __init__(self, repetitions, axis="T", **kwargs):
+    """
+    :param LayerBase repetitions: number of repetitions for each sequence and position in target axis, [B,T] or [T,B]
+    :param str|int axis: (dynamic) axis for repetition (currently only time axis is supported)
+    """
+    super(RepeatLayer, self).__init__(**kwargs)
+    repetitions_data = repetitions.output
+    input_data = self.input_data
+    assert self.input_data
+    original_axis = input_data.get_axis_from_description(axis)
+    if original_axis < input_data.batch_dim_axis:
+      input_data = input_data.copy_as_batch_major().copy_move_axis(original_axis+1, 1)
+    else:
+      input_data = input_data.copy_as_batch_major().copy_move_axis(original_axis, 1)
+    repetitions_data = repetitions_data.copy_as_batch_major()
+    # pad the target axis
+    paddings = [(0, 1) if i == 1 else (0, 0) for i in range(input_data.batch_ndim)]
+    padded_data = tf.pad(input_data.placeholder, paddings)  # [B, T+1, ...]
+    # those are the sequence lengths after expansion
+    target_seq_len = tf.reduce_sum(repetitions_data.placeholder, axis=1)  # [B], the new size_placeholder for 'T'
+    # maximum sequence length after expansion
+    max_duration = tf.reduce_max(target_seq_len)  # [] == T'
+    # the new padding is the difference of the maximum to the new target
+    target_padding_steps = tf.expand_dims(max_duration - target_seq_len, 1)  # [B, 1]
+    # add the repetitions for the artificial padding position
+    target_repetitions = tf.concat([repetitions_data.placeholder, target_padding_steps], axis=1)  # [B, T+1]
+    # flatten batch and time
+    shape = tf.shape(padded_data)
```

Comment:
Use `tf_util.get_shape` instead.
```suggestion
    shape = tf_util.get_shape(padded_data)
```